CRM/Remoteevent/RegistrationProfile.php

```diff
@@ -263,69 +274,72 @@ public static function validateProfileData($validationEvent)
      */
     public static function getRegistrationProfile($profile_name)
     {
-        $profiles = self::getAvailableRegistrationProfiles('name');
-        if (in_array($profile_name, $profiles)) {
-            // get class
-            $class_candidate = "CRM_Remoteevent_RegistrationProfile_{$profile_name}";
-            if (class_exists($class_candidate)) {
-                return new $class_candidate();
-            } else {
-                // todo: extend to use Symfony hooks
-                throw new Exception(E::ts("Implementation for profile '%1' not found.", [1 => $profile_name]));
-            }
-        } else {
-            throw new Exception(E::ts("Registration profile '%1' is not available (any more).", [1 => $profile_name]));
-        }
+        $profile_list = new RemoteEvent\Event\RegistrationProfileListEvent();
+        // dispatch Registration Profile Event and try to instantiate a profile class from $profile_name
+        Civi::dispatcher()->dispatch('civi.remoteevent.registration.profile.list', $profile_list);
+
+        return $profile_list->getProfileInstance($profile_name);
```

Comment:
There should be something like this, yes. 

But since, as @pbatroff pointed out, this is already the case (albeit by deactivating an extension), we should create a separate ticket about it and don't solve this in here.

Reply:
https://github.com/systopia/de.systopia.remoteevent/issues/43